pkg/storaged/btrfs/subvolume.jsx

```diff
@@ -0,0 +1,129 @@
+    mounting_dialog(client, block, "mount", null, subvol);
+}
+
+export function make_btrfs_subvolume_page(parent, volume, subvol) {
+    const [mount_point, is_mounted] = subvolume_mount_info(volume, subvol);
+    let location = "";
+    const use = btrfs_usage(client, volume);
+    const actions = [];
+
+    const backing_block = client.blocks[volume.path];
+    const content_block = client.blocks_fsys[volume.path];
+    const fstab_config = get_fstab_config_with_client(client, backing_block, false, subvol);
+    const mismount_warning = check_mismounted_fsys(backing_block, content_block, fstab_config, subvol);
+
+    if (mount_point) {
+        location = mount_point;
+    }
+
+    if (is_mounted) {
+        actions.push({
+            title: _("Unmount"),
+            action: () => subvolume_unmount(volume, subvol),
+        });
+    } else {
+        actions.push({
+            title: _("Mount"),
+            action: () => subvolume_mount(volume, subvol),
+        });
+        if (mount_point) {
+            location += " " + _("(not mounted)");
+        }
+    }
+
+    // Don't allow deleting the root subvolume
+    // TODO: check if the root subvolume is always or if this requires `btrfs subvolume get-default`?
+    if (subvol.id !== 5) {
+        // TODO in follow up
+        // actions.push({
+        //     title: _("Delete"),
+        //     action: () => subvolume_delete(volume, subvol),
+        //     danger: true,
+        // });
+    }
+
+    const card = new_card({
+        title: _("btrfs subvolume"),
+        next: null,
+        // HACK: page_location does not work well with a path that contains a Number, likely due to JSON.stringify usage?
+        page_location: ["btrfs", volume.data.uuid, String(subvol.id)],
+        page_name: subvol.pathname,
+        page_size: is_mounted && <StorageUsageBar stats={use} short />,
+        location,
+        component: BtrfsSubvolumeCard,
+        has_warning: !!mismount_warning,
+        props: { volume, subvol, mount_point, is_mounted, mismount_warning, backing_block, content_block, fstab_config },
+        actions,
+    });
+    new_page(parent, card);
+}
+
+const BtrfsSubvolumeCard = ({ card, volume, subvol, mount_point, is_mounted, backing_block, content_block, fstab_config, mismount_warning }) => {
+    return (
+        <StorageCard card={card} alert={mismount_warning &&
+        <MismountAlert warning={mismount_warning}
+                                    fstab_config={fstab_config}
+                                    backing_block={backing_block} content_block={content_block} subvol={subvol} />}>
+            <CardBody>
+                <DescriptionList className="pf-m-horizontal-on-sm">
+                    <StorageDescription title={_("Name")} value={subvol.pathname} />
+                    <StorageDescription title={_("ID")} value={subvol.id} />
+                    {mount_point &&
+                        <StorageDescription title={_("Mount point")} value={mount_point} />
```

Comment:
These 12 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19690-20231205-210454-49836fdb-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)